core/src/main/scala/org/apache/spark/sql/delta/optimizablePartitionExpressions.scala

```diff
@@ -547,6 +546,49 @@ case class DateFormatPartitionExpr(
   }
 }
 
+/** The rules for the generation expression `date_trunc(field, col)`. */
+case class DateTruncPartitionExpr(format: String, partitionColumn: String)
```

Comment:
I think it's better to name this `TimestampTruncPartitionExpr` since there is also `TruncDate`

Reply:
Good point, I was so focused on the date_trunc call, hence the name, but this way its way clearer